src/cli/api/local_api.py

```diff
@@ -123,19 +127,122 @@ def list_applications(self) -> List[ApplicationType]:
         return local_applications
 
     # Todo: Update confluence scenario diagram since application_unique() and structure_valid() are swapped
-    def is_application_valid(self, application: str) -> Tuple[bool, str]:
-        if self.__is_application_unique(application) and \
-           self.__is_structure_valid(application) and \
-           self.__is_config_valid(application):
-            return True, "Valid"
-
-        return False, "Invalid"
-
-    def __is_structure_valid(self, application: str) -> bool:
-        pass
+    def is_application_valid(self, application: str) -> List[str]:
+        """
+        Function that checks if:
+        - The application is valid by validating if it exists in .qne/application.json
+        - The file and directory structure is correct
+        - The json files in the config directory contain valid json
+        - The json files passes against schema validation
+
+        returns:
+            Returns empty list when all validations passes
+            Returns list containing error messages of the validations that failed
+        """
+        error_list = []
+        is_unique, _ = self.__is_application_unique(application)
 
-    def __is_config_valid(self, application: str) -> bool:
-        pass
+        if is_unique:
+            error_list.append("Application does not exist")
+        for items in self.__is_structure_valid(application):
+            error_list.append(items)
+        for items in self.__is_config_valid(application):
+            error_list.append(items)
+
+        # Remove any duplicates from list
+        filtered_error_list = []
+        for i in error_list:
+            if i not in filtered_error_list:
+                filtered_error_list.append(i)
+
+        return filtered_error_list
+
+    def __is_config_valid(self, application: str) -> List[str]:
+        # Validate if json string is correct and validate against json schema's
+        schema_app_path = Path(BASE_DIR + "/schema/applications")
+        app_config = Path(self.__config_manager.get_application(application)['path']) / "config"
```

Comment:
With using 'path' you are exposing structure data that should only be (hidden) in config manager. When we change the configuration file to use 'directory' instead of 'path' this fails.
Why not add a method get_application_path(application_name) to config manager.
reuse this method in __is_structure_valid and get_application_config and maybe more places